x-pack/plugins/security/public/components/breadcrumb.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, useEffect, useRef, useContext, FunctionComponent } from 'react';
+import { EuiBreadcrumb } from '@elastic/eui';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+interface BreadcrumbsContext {
+  parents: BreadcrumbProps[];
+  onMount(breadcrumbs: BreadcrumbProps[]): void;
+  onUnmount(breadcrumbs: BreadcrumbProps[]): void;
+}
+
+const BreadcrumbsContext = createContext<BreadcrumbsContext | undefined>(undefined);
+
+export interface BreadcrumbProps extends EuiBreadcrumb {
+  text: string;
+}
+
+/**
+ * Component that automatically sets breadcrumbs and document title based on the render tree.
+ *
+ * @example
+ * // Breadcrumbs will be set to: "Users > Create"
+ * // Document title will be set to: "Create - Users"
```

Comment:
Do we actually want the doc title here to have the full breadcrumb trail? @ryankeairns ?

Reply:
It looks like we have a mix of things going on currently. The majority of views seem to only show 'Elastic', which is not ideal, while others plugins provide their name in the title (e.g. _Rollup Jobs - Elastic_). 

There are only a couple that seem to go beyond the top-level plugin name (e.g. _Add repository - Snapshot and Restore - Elastic_). On the one hand, I can see the apparent clarity of adding more detail, on the other, the title attributes could get rather long as things go deeper across the various plugins.

My inclination is to follow a limited _<current page title/last breadcrumb> - <plugin name> - Elastic_ pattern. Then, at most, you have three parts to the title regardless of depth. In this case, the title would then be 'Create - Users - Elastic'

Reply:
Thanks for taking the time to review Caroline and Ryan - Much appreciated. 

In regards to document title, I don't think the length will grow unreasonable here. Tabs are fairly small, especially when users have lots of them open so it's expected that they get truncated. As an example, even the document title of this pull request on Github get's truncated for me in both Safari and Chrome. 

What's more important is that the document title helps users identify the page and that the most relevant information is located at the beginning with more generic bits at the end.

If you think about a scenario where an administrator opens multiple users in separate tabs to manage them seeing 4 tabs with the same title "Users - Elastic" isn't going to be very helpful. Having the actual document (in this case the username) will make it easy to identify the correct page: 

<img width="1139" alt="Screenshot 2021-01-05 at 10 22 08" src="https://user-images.githubusercontent.com/190132/103635782-eb5a5880-4f40-11eb-8872-2917cee6d081.png">

Reply:
I think the worry is mostly about what happens if there are 5+ breadcrumb items. The document title becomes extremely long, and then if you add on the time it takes screen readers to read each tab it's more exacerbated. I think what @ryankeairns is proposing is just to ensure limiting the document title to on the last 2 breadcrumbs **plus** `Elastic`. So you'd definitely get as you are showing in your screenshot with `caroline - Users - Elastic` but it would never get longer than that.

Reply:
Got you, I'll limit these to two breadcrumbs then.